virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -80,3 +80,43 @@ def create_primitive_disk_xml(type_name, disk_device, device_target, device_bus,
         disk_xml.auth = disk_xml.new_auth(**disk_auth_dict)
     logging.debug("new disk xml in create_primitive_disk is: %s", disk_xml)
     return disk_xml
+
+
+def create_custom_metadata_disk(disk_path, disk_format,
+                                disk_device, device_target, device_bus, max_size, disk_inst=None):
+    """
+    Create another disk for a given path,customize driver metadata attribute
+
+    :param disk_path: the path of disk
+    :param disk_format: the format to disk image
+    :param disk_device: the disk device type
+    :param device_target: the target of disk
+    :param device_bus: device bus
+    :param max_size: metadata_cache max size
+    :param disk_inst: disk instance
+    :return: disk object if created or updated successfully
+    """
+    if disk_inst:
+        custom_disk = disk_inst
+    else:
+        custom_disk = Disk(type_name='file')
+    if disk_device:
+        custom_disk.device = disk_device
+    source_dict = {}
+    if disk_path:
+        source_dict.update({'file': disk_path})
+    custom_disk.source = custom_disk.new_disk_source(
+        **{"attrs": source_dict})
+    if device_target:
```

Comment:
For the above 2 params, disk_path & device_target, are they must be given? If not, I think the default value in function params should be given with like disk_path=None; If not, then I think no need to judge "if device_target" exist.

Reply:
@kylazhang 
For snapshotdisk, device_target is not mandatory
For specific device, such as cdrom , inner part in <source  /source> can be  empty

Reply:
Then if it is not necessary, please add "disk_path=None, device_target=None" in function params

Reply:
And also I think it is a better way but your way is not unacceptable after we discuss and think more. It depends on you finally.